scripts/check_for_docs_deps_changes

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+CHANGED_FILES=$(git diff HEAD origin/develop --name-only)
+printf '%s\n' "[CHANGED FILES]" "${CHANGED_FILES[@]}" ""
+
+DEPENDENCIES=$(python scripts/trace_docs_deps.py docs)
+printf '%s\n' "[DEPENDENCIES]" "${DEPENDENCIES[@]}" ""
+
+echo "Checking for file changes from 'great_expectations/' that impact 'docs/'..."
+for FILE in ${DEPENDENCIES}; do
+    if [[ ${CHANGED_FILES[@]} =~ $FILE ]]; then
+        echo "Found change in local dependency:" $FILE
+        echo "##vso[task.setvariable variable=DocsDependenciesChanged;isOutput=true]true"
+    fi
+done
```

Comment:
@alexsherstinsky I've added the styling you requested after moving this script to a separate file.

I've got a question about this: `CHANGED_FILES=$(git diff HEAD origin/develop --name-only)`. Should this be `$(git diff HEAD HEAD~ --name-only)`? 

If we're writing an extensive PR that has a single change in one of these dependencies, every single commit in that PR will trigger the pipeline (due to `diff` evaluating changes between branches and not just changes from the most recent commit). Is there a mechanism in Azure that kills pipelines early if we stack up too many?